File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(a) {
    let i =0;
    let j =1;
    let k =1;
    let n =a.length;
    while (j<n){
        if(a[i]==a[j])
            j++;
        else{
            a[++i]=a[j++];
            k++;
        }
    }
    return k;
};

File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.java
class Solution {
    public void swap(int[] nums, int i , int j){
        int temp = nums[i];
        nums[i]=nums[j];
        nums[j]=temp;
    }
    public int removeDuplicates(int[] nums) {
        int i = 0;
        int j = 1;
        int n = nums.length;

        while(j<n){
            if(nums[i]!=nums[j]){
                i++;
                swap(nums,i,j);
                j++;
            }
            else j++;
        }
        return i+1;
    }
}